plugins/modules/foreman_role.py

```diff
@@ -101,7 +118,21 @@ def main():
         if 'organizations' in entity_dict:
             entity_dict['organizations'] = module.find_resources_by_name('organizations', entity_dict['organizations'], thin=True)
 
-    changed = module.ensure_entity_state('roles', entity_dict, entity)
+    filters = entity_dict.pop("filters", None)
+
+    changed, entity = module.ensure_entity('roles', entity_dict, entity)
+
+    if not module.desired_absent:
+        if filters is not None:
+            existing_filters = entity['filters']
```

Comment:
why are you assigning this? just for nicer `for old_filter in existing_filters` later? or do you meant to use `.copy()`?

Reply:
Good question. Now I realize that entity won't be reloaded with new filters, so this variable is not really needed.